### DISCUSSION TITLE
Trying to Override the default layout using renderView function with jQuery .dialog

### DISCUSSION THREAD

**Discussion by cfcoder** [wheels-dev/wheels#1354] [Q&A]
Trying to open a jQuery .dialog that contains the content of a .cfm file using Wheels.  I initialize the dialog in a click handler. Then I try to load the file using urlFor() function to provide the URL to load.

In the parent .cfm file, I have:
```
<div id="dialog">
</div>
```

```
$("#create_project").on("click", function(){
	// initialize dialog
	var dialog1 = $("#dialog").dialog({ 
		autoOpen: false,
		height: 600,
		width: 850
	});

	// load content and open dialog
	dialog1.load('<cfoutput>#urlFor(route="My_Create_Project", onlyPath=false)#</cfoutput>').dialog('open');

	// NOTE: in the rendered source the above line becomes:
	// dialog1.load('http://dev.localhost:60172/projects/create').dialog('open');
});
```

In the controller, the last directive in the create() function is:
`renderView(layout=false);`

If I call that route in a browser window, it works as expected. But when called from the load function, it opens the dialog but the file is picking up the default "layout.cfm" in the top view folder. I have specified for the action function being called to renderView with layout=false. But it appears to be ignored.

What am I missing here?

**Comment by chapmandu** [comment]
I'm guessing that your jquery load function is using the http GET method. In general the wheels create function normally uses the POST method so I'd guess you're actually calling the index function. Have a look at the routes tab to see which actions use which methods.

**Reply by cfcoder** [reply]
Actually, the action I'm loading is using a POST method, not the GET, referencing the action by name attribute value. The route uses a pattern of "projects/create" and to="projects##create".  No index function is being called AFAIK.

**Reply by cfcoder** [reply]
Maybe I misunderstood you. I just changed that POST to be a GET and now the load comes up with just the create.cfm template content without the default layout.cfm wrapper.  Please clarify what you meant.

**Reply by chapmandu** [reply]
Can you post the contents of routes.cfm here.

**Reply by cfcoder** [reply]
```
mapper()
	...
	.post(name="Create_Project", pattern="projects/create", to="projects##create")
	.post(name="New_Project", pattern="projects/new", to="projects##new")
	.get(name="Open_Project", pattern="projects/open/[key]", to="projects##open")
	...
	.resources(name="projects")
	.wildcard()
	.root(to="session##new", method="get")
.end();

```

**Reply by cfcoder** [reply]
I realize this isn't following proper Wheels routing, in that it is defining custom routes instead of relying on the .resources to create all the CRUD you need. But I didn't write this, just trying to work with it.

**Reply by chapmandu** [reply]
Your routes look okay. I still think jquery is issuing a GET request here. 

dialog1.load('<cfoutput>#urlFor(route="My_Create_Project", onlyPath=false)#</cfoutput>')

Check your browser's network tab to see what's happening.
